src/Controls/src/Core/Handlers/Items/iOS/ItemsViewController.cs

```diff
@@ -175,9 +177,59 @@ public override void ViewWillLayoutSubviews()
 		{
 			ConstrainToItemsView();
 			base.ViewWillLayoutSubviews();
+			InvalidateMeasureIfContentSizeChanged();
 			LayoutEmptyView();
 		}
 
+		void InvalidateMeasureIfContentSizeChanged()
+		{
+			var contentSize = CollectionView.CollectionViewLayout.CollectionViewContentSize;
+
+			if (_initialized && (_previousContentSize.Width != contentSize.Width
+				|| _previousContentSize.Height != contentSize.Height))
+			{
+				var screenFrame = CollectionView.Window.Frame;
+				var screenWidth = screenFrame.Width;
+				var screenHeight = screenFrame.Height;
+				bool invalidate = false;
+
+				// If both the previous content size and the current content size are larger
+				// than the screen size, then we know that we're already maxed out and the 
+				// CollectionView items are scrollable. There's no reason to force an invalidation
+				// of the CollectionView to expand/contract it.
+
+				// If either size is smaller than that, we need to invalidate to ensure that the 
+				// CollectionView is re-measured and set to the correct size.
+
+				if (contentSize.Width < screenWidth || contentSize.Width < screenWidth)
```

Comment:
Should these take into account what comparison on line 188 caused you to get here?

I'm testing this in the controls gallery and the if statement on line 188 is true because

```C#
// 2708 != 2820
_previousContentSize.Height != contentSize.Height
```

But the invalidate gets triggered because contentSize width is always less than the screenWidth

```C#
// 418 < 430  
contentSize.Width < screenWidth
```

Should the if be

```C#
if (_previousContentSize.Width != contentSize.Width && (contentSize.Width < screenWidth || contentSize.Width < screenWidth))
```

Reply:
Fixed, thanks!